src/solvers/refinement/string_constraint_generator.cpp

```diff
@@ -0,0 +1,2145 @@
+/*******************************************************************\
+
+Module: Generates string constraints to link results from string functions
+        with their arguments. This is inspired by the PASS paper at HVC'13
+	which gives examples of constraints for several functions.
```

Comment:
Yes, added it https://github.com/diffblue/cbmc/pull/374/commits/76ea888448a129beecb45b95dae0b778d8bd855a